altinn/src/test/kotlin/no/nav/helsearbeidsgiver/inntektsmelding/altinn/AltinnAppTest.kt

```diff
@@ -0,0 +1,112 @@
+import no.nav.helsearbeidsgiver.felles.json.toJson
+import no.nav.helsearbeidsgiver.felles.json.toMap
+import no.nav.helsearbeidsgiver.felles.test.rapidsrivers.firstMessage
+import no.nav.helsearbeidsgiver.felles.test.rapidsrivers.sendJson
+import no.nav.helsearbeidsgiver.inntektsmelding.altinn.Mock.altinnOrganisasjoner
+import no.nav.helsearbeidsgiver.inntektsmelding.altinn.Mock.innkommendeMelding
+import no.nav.helsearbeidsgiver.inntektsmelding.altinn.Mock.toMap
+import no.nav.helsearbeidsgiver.maskinporten.TokenResponse
+import no.nav.helsearbeidsgiver.utils.json.serializer.set
+import no.nav.helsearbeidsgiver.utils.json.toJson
+import okhttp3.mockwebserver.MockResponse
+import okhttp3.mockwebserver.MockWebServer
+import org.junit.jupiter.api.assertDoesNotThrow
+
+class AltinnAppTest :
+    FunSpec({
+        val testRapid = TestRapid()
+        lateinit var server: MockWebServer
+        beforeEach {
+            server = MockWebServer()
+            server.start()
+            val baseUrl = server.url("/").toString()
+            println("baseUrl== $baseUrl")
+        }
+        afterEach { clearAllMocks() }
+
+        test("tester at Altinn client  og maskinporten kaller riktig endepunkt og sender riktig data") {
+            mockkObject(RapidApplication)
+            every { RapidApplication.create(any()) } returns testRapid
+
+            val maskinportenToken = TokenResponse("test_token", "Bearer", 3600, "test:test1")
+            val tokenResponse = Json.encodeToString(TokenResponse.serializer(), maskinportenToken)
+            server.enqueue(
+                MockResponse()
+                    .setBody(tokenResponse)
+                    .addHeader("Content-Type", "application/json"),
+            )
+            val altinnResponse = Json.encodeToString(altinnOrganisasjoner)
+            val mockResponse =
+                MockResponse()
+                    .setBody(altinnResponse)
+                    .addHeader("Content-Type", "application/json")
+            server.enqueue(mockResponse)
+
+            mockEnv(server)
+
+            assertDoesNotThrow { main() }
+
+            val innkommendeMelding = innkommendeMelding()
+
+            testRapid.sendJson(innkommendeMelding.toMap())
+
+            testRapid.inspektør.size shouldBeExactly 1
+
+            val altinnOrgnr =
+                altinnOrganisasjoner
+                    .mapNotNull { it.orgnr }
+                    .toSet()
+
+            testRapid.firstMessage().toMap() shouldContainExactly
+                mapOf(
+                    Key.EVENT_NAME to innkommendeMelding.eventName.toJson(),
+                    Key.UUID to innkommendeMelding.transaksjonId.toJson(),
+                    Key.DATA to
+                        innkommendeMelding.data
+                            .plus(
+                                Key.ORG_RETTIGHETER to altinnOrgnr.toJson(String.serializer().set()),
+                            ).toJson(),
+                )
```

Comment:
Er dette relevant? Tester man ikke klientkonfigen her? For meg så ser dette ut som duplisert testkode.

Reply:
Denne testen kjører hele klienten og mocker HTTP-kallet.
Den utfører kun verifisering og gjenbruker testdata fra AltinnRiverTest.

Reply:
Men holder det ikke å sjekke at HTTP-kallet er gjort?